### PR TITLE
Fix twisted.python.filepath.setContent() on Windows with Python 3

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/8610

### CODE REVIEW

**Comment by rodrigc** [comment]
@hawkowl please review, this is cherry-picked from https://github.com/twisted/twisted/blob/moar-windows-8025-7/twisted/python/filepath.py on your branch

**Comment by hawkowl** [comment]
@rodrigc I can't review this, since I wrote the original code :)

paging @glyph

**Comment by glyph** [comment]
This _looks_ correct to me, but is there a skipped test someplace that could be un-skipped, or should a new test be written?

**Comment by rodrigc** [comment]
@glyph without this fix, many, many tests fail on Windows with Python 3
because they call FilePath.setContent().
This is one step towards getting a usable Windows Python 3 buildbot or appveyor.

**Comment by glyph** [comment]
@rodrigc Gotcha.

**Comment by glyph** [comment]
Should the ticket here be in review?

**Comment by rodrigc** [comment]
I put it in review

File: twisted/python/filepath.py [hunk 1]
```diff
@@ -1529,8 +1529,8 @@ def setContent(self, content, ext=b'.new'):
         with sib.open('w') as f:
             f.write(content)
         if platform.isWindows() and exists(self.path):
-            os.unlink(self.path)
-        os.rename(sib.path, self.path)
+            os.remove(self.path)
```
Comment (Lukasa) [inline twisted/python/filepath.py]: What's the reasoning for this change? It seems like needless churn, but it's also harmless, so I'm not going to block review on it. Generally speaking it'd be nice if we didn't make a change like this, so please consider removing it.

Comment (rodrigc) [inline twisted/python/filepath.py]: That's not a big deal, I'll change it back.  I took this patch from hawkowl's branch

**Pull request merged**